File: OPENAPI-REST-API/swagger-client/scala-akka-http-server/src/main/scala/io/swagger/server/model/ServerOrderPostRequest.scala
package io.swagger.server.model


/**
 * Request body to place a custom dedicated server order. All ids come from getNewServer's config_li / regions.
 *
 * @param cpu CPU configuration id from config_li.cpu_li. Constrains valid hd/memory options.
 * @param hd Hard-drive configuration id(s) from config_li.hd_li[cpu]. Accepts a single id or an array of ids.
 * @param memory Memory configuration id from config_li.memory_li[cpu].
 * @param bandwidth Bandwidth configuration id from config_li.bandwidth_li.
 * @param ips IP block configuration id from config_li.ips_li.
 * @param os Operating System configuration id from config_li.os_li.
 * @param cp Control Panel configuration id from config_li.cp_li.
 * @param raid RAID configuration id from config_li.raid_li.
 * @param region Region id from the regions list.
 * @param servername Server hostname. Must pass valid_hostname().
 * @param rootpass Root password for the server.
 * @param tos Terms-of-service acceptance. Must be true to place the order.
 * @param comment Optional free-form order comment.
 */
case class ServerOrderPostRequest (
  cpu: Int,
  hd: OneOfServerOrderPostRequestHd,
  memory: Int,
  bandwidth: Int,
  ips: Int,
  os: Int,
  cp: Int,
  raid: Int,
  region: Int,
  servername: String,
  rootpass: String,
  tos: Boolean,
  comment: Option[String]
)

File: OPENAPI-REST-API/openapi-client/javascript-closure-angular/API/Client/ServerOrderPostRequest.js
goog.provide('API.Client.ServerOrderPostRequest');

/**
 * Request body to place a custom dedicated server order. All ids come from getNewServer's config_li / regions.
 * @record
 */
API.Client.ServerOrderPostRequest = function() {}

/**
 * CPU configuration id from config_li.cpu_li. Constrains valid hd/memory options.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.cpu;

/**
 * @type {!API.Client.ServerOrderPostRequest_hd}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.hd;

/**
 * Memory configuration id from config_li.memory_li[cpu].
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.memory;

/**
 * Bandwidth configuration id from config_li.bandwidth_li.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.bandwidth;

/**
 * IP block configuration id from config_li.ips_li.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.ips;

/**
 * Operating System configuration id from config_li.os_li.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.os;

/**
 * Control Panel configuration id from config_li.cp_li.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.cp;

/**
 * RAID configuration id from config_li.raid_li.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.raid;

/**
 * Region id from the regions list.
 * @type {!number}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.region;

/**
 * Server hostname. Must pass valid_hostname().
 * @type {!string}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.servername;

/**
 * Root password for the server.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.rootpass;

/**
 * Terms-of-service acceptance. Must be true to place the order.
 * @type {!boolean}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.tos;

/**
 * Optional free-form order comment.
 * @type {!string}
 * @export
 */
API.Client.ServerOrderPostRequest.prototype.comment;

